GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -1006,91 +1007,121 @@ protected void invokeUrl(String serverBase, String urlString,
-                    if (!isSupportedReportFormat(firstQuestionColumnIndex)) {
-                        errorMap.put(firstQuestionColumnIndex,
-                                "Found the first question at the wrong column index");
-                        break;
-                    }
-                    log.info("Importing report with first question column index: "
-                            + firstQuestionColumnIndex);
-                }
-            }
+        return questionCount;
+    }
 
-            if (!firstQuestionFound) {
-                errorMap.put(-1, "A question could not be found");
-            }
+    /*
+     * validate
+     * Called from Clojure code before executeImport()
+     */
+    @Override
+    public Map<Integer, String> validate(File file) {
+        Map<Integer, String> errorMap = new HashMap<Integer, String>();
 
-            Workbook wb = sheet.getWorkbook();
+        try {
+            Workbook wb = getDataSheet(file).getWorkbook();
 
-            //check that all mandatory columns exist on all sheets
+            //check each sheet in turn
+            int questions = 0;
             for (int i = 0; i < wb.getNumberOfSheets(); i++) {
-                sheet = wb.getSheetAt(i);
-                Map<String, Integer> index = getMetadataColumnIndex(sheet, firstQuestionColumnIndex, headerRowIndex,
-                        false);
-                if (!checkCol(index, DATAPOINT_IDENTIFIER_COLUMN_KEY)) {
-                    errorMap.put(-1, "Column header '" + IDENTIFIER_LABEL + "' missing on sheet " + i);
-                }
-                if (!checkCol(index, DATAPOINT_NAME_COLUMN_KEY)) {
-                    errorMap.put(-1, "Column header '" + DISPLAY_NAME_LABEL + "' missing on sheet " + i);
-                }
-                if (!checkCol(index, SURVEY_INSTANCE_COLUMN_KEY)) {
-                    errorMap.put(-1, "Column header '" + INSTANCE_LABEL + "' missing on sheet " + i);
-                }
-                if (!checkCol(index, COLLECTION_DATE_COLUMN_KEY)) {
-                    errorMap.put(-1, "Column header '" + SUB_DATE_LABEL + "' missing on sheet " + i);
-                }
-                if (!checkCol(index, SUBMITTER_COLUMN_KEY)) {
-                    errorMap.put(-1, "Column header '" + SUBMITTER_LABEL + "' missing on sheet " + i);
-                }
-                if (!checkCol(index, DURATION_COLUMN_KEY)) {
-                    errorMap.put(-1, "Column header '" + DURATION_LABEL + "' missing on sheet " + i);
-                }
-                if (i > 0 && !checkCol(index, REPEAT_LABEL)) {
-                    errorMap.put(-1, "Column header '" + REPEAT_LABEL + "' missing on sheet " + i);
+                Sheet sheet = wb.getSheetAt(i);
+                String name = sheet.getSheetName();
+                if (i == 0) { //TODO OR just I==0??
+                //TODO possible variant:   if (RAW_DATA_SHEET_LABEL.equals(name)) {
```

Comment:
In fact I think here it should be enough to have something like
```java
int isBaseSheet = false;
if (label == "Raw data") isBaseSheet = true;
else if (label matches "Group ...") isBaseSheet ...
```

And a sheet which doesnt match either of the two doesnt necessarily need to be an error but we ignore it and dont consider for the import.

Reply:
The Clojure validation only validates question ID's on sheet 0, right now, and the import also assumes base sheet to be 0, so I want to match that until we redo all this.
Unknown-name sheets now ignored for verification.